solarforecastarbiter/metrics/tests/test_calculator.py

```diff
@@ -412,6 +413,19 @@ def test_calculate_deterministic_metrics_no_metrics(
         )
 
 
+def test_calculate_deterministic_metrics_no_cost_param(
+        create_processed_fxobs, single_forecast_observation):
+    proc_fx_obs = create_processed_fxobs(single_forecast_observation,
+                                         np.array([1]), np.array([1])).replace(
+                                             cost=None)
+
+    out = calculator.calculate_deterministic_metrics(
+            proc_fx_obs, LIST_OF_CATEGORIES, ['cost']
+    )
+    for o in out.values:
+        assert np.isnan(o.value)
+
+
```

Comment:
I think we need a new test or a modified test to check that costs are correctly calculated through `calculator.py`.

Reply:
sure. It is included in the current tests (because DETERMINISTIC_METRICS includes it), but not that the specific value is correct